website/docs/plugin/best-practices/index.mdx

```diff
@@ -0,0 +1,39 @@
+---
+page_title: Plugin Development - Best Practices
+description: >-
+  Patterns that ensure a consistent user experience, including naming and versioning.
+---
+
+# Terraform Plugin Best Practices
```

Comment:
This was copied from the original index, should we add more blurbs for each of the remaining pages or were they left out intentionally?

- Interacting with providers
- Other languages
- Sensitive state

Reply:
Added a first draft for them 😄

Reply:
These look good!